pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -119,11 +122,13 @@ private QueryContext(String tableName, List<ExpressionContext> selectExpressions
       @Nullable FilterContext filter, @Nullable List<ExpressionContext> groupByExpressions,
       @Nullable FilterContext havingFilter, @Nullable List<OrderByExpressionContext> orderByExpressions, int limit,
       int offset, Map<String, String> queryOptions, @Nullable Map<String, String> debugOptions,
-      BrokerRequest brokerRequest) {
+      BrokerRequest brokerRequest, boolean hasFilteredAggregations,
```

Comment:
`hasFilteredAggregations` should not be set through the constructor. It is updated in `generateAggregationFunctions()`

Reply:
Good catch, fixed.